NitroxClient/GameLogic/Vehicles.cs

```diff
@@ -106,5 +120,170 @@ private void OnVehiclePrefabLoaded(GameObject prefab, string guid, Vector3 spawn
 
             GuidHelper.SetNewGuid(gameObject, guid);
         }
+
+        public void DestroyVehicle(VehicleRemoveEntry entry, PlayerManager remotePlayerManager) //Destroy Vehicle From network
+        {
+            Optional<GameObject> Object = GuidHelper.GetObjectFrom(entry.Guid);
+            if (Object.IsPresent())
+            {
+                GameObject T = Object.Get();
+                Vehicle vehicle = T.RequireComponent<Vehicle>();
+
+                if (entry.GetPilotingMode) //Check Remote Object Have Player inside
+                {
+                    Optional<RemotePlayer> player = remotePlayerManager.Find(vehicle.pilotId);
+
+                    if (player.IsPresent()) // Get Remote Player Inside == vehicle.pilotId  Remove From Vehicle Before Destroy
+                    {
+                        RemotePlayer playerInstance = player.Get();
+                        playerInstance.SetVehicle(null);
+                        playerInstance.SetSubRoot(null);
+                        playerInstance.SetPilotingChair(null);
+                        playerInstance.AnimationController.UpdatePlayerAnimations = true;
+                    }
+                }
+
+                if (vehicle.GetPilotingMode()) //Check Local Object Have Player inside
+                {
+                    vehicle.ReflectionCall("OnPilotModeEnd", false, false, null);
+                    if (!Player.main.ToNormalMode(true))
+                    {
+                        Player.main.ToNormalMode(false);
+                        Player.main.transform.parent = null;
+                    }
+                }
+
+                if (vehicle.gameObject != null) // Destroy Vehicle
+                {
+                    if (vehicle.destructionEffect)
+                    {
+                        GameObject gameObject = UnityEngine.Object.Instantiate<GameObject>(vehicle.destructionEffect);
+                        gameObject.transform.position = vehicle.transform.position;
+                        gameObject.transform.rotation = vehicle.transform.rotation;
+                    }
+                    UnityEngine.Object.Destroy(vehicle.gameObject);
+                }
+            }
+        }
+
+        public void Add(Vehicle vehicle) //Add Vehicle Fron Network Packet
```

Comment:
Let's rename this to CreateNewVehicle and make a new method called BroadcastNewVehicle.  The new method, BroadcastNewVehicle can call CreateNewVehicle then send the packet.  This would allow you to call only CreateNewVehicle without having to do packet surpression in the PacketProcessor. 

We should do the same thing for RemoveVehicle as well.